samples/System.Numerics.Tensors.Samples/TensorSamples/WrapNativeMemory.cs

```diff
@@ -0,0 +1,89 @@
+﻿using System;
+using System.Numerics;
+using System.Runtime.InteropServices;
+using System.Text;
+
+namespace TensorSamples
+{
+    partial class WrapNativeMemory
+    {
+        public static void RunSample()
+        {
+            Console.WriteLine($"*** {nameof(WrapNativeMemory)} ***");
+            
+            Console.WriteLine($"Allocated as part of the operation");
+            // in this case we pretend that we cannot know the size of the tensor up front and require the native library to allocate it and tell us the size.
+            var multTable = GetMultiplicationTable(5);
+
+            Console.WriteLine("Multiplication table:");
+            Console.WriteLine(multTable.GetArrayString());
+
+            Console.WriteLine("Sums:");
+            for (int row = 0; row < multTable.Dimensions[0]; row++)
+            {
+                Console.WriteLine(GetRowSum(multTable, row));
+            }
+            
+            // the memory will be freed when the GC decides to collect the Tensor, we can *try* to force it but no garuntee.
+            multTable = null;
+            Console.WriteLine("Forcing GC.");
+            GC.Collect();
+            GC.WaitForPendingFinalizers();
+
+            Console.WriteLine($"Allocated prior to the operation");
+            // in this case we'll assume we can know the size of the buffer up front and manage the lifetime explicitly
+            using (var nativeMemory = NativeMemory<double>.Allocate(5 * 5))
+            {
+                Span<int> dimensions = stackalloc int[2];
+                dimensions[0] = dimensions[1] = 5;
+                var tensor = new DenseTensor<double>(nativeMemory.Memory, dimensions);
```

Comment:
Since `DenseTensor` is a reference type, this mean an allocation for every time native memory is "wrapped"... have value type `TensorView`s (ala `Span`) been considered to avoid this?